File: 2022/lua/d1_1.lua

most = 0
elf_load = 0
for line in io.lines("/home/joe/dev/advent-of-code/2022/python/day1/input.txt") do
    if line ~= '' then
	elf_load = elf_load + tonumber(line)
    else
	if elf_load > most then
	    most = elf_load
	end
	elf_load = 0
    end
    print(elf_load)
end
print(most)

File: 2022/rust/day1/d1_1.rs

fn main() {
    let contents = std::fs::read_to_string("/home/joe/dev/advent-of-code/2022/python/day1/input.txt")
        .expect("Should have been able to read the file");
    let trimmed = contents.trim_end();

    let str_loads: Vec<&str> = trimmed.split("\n\n").collect();
    let mut most: u32 = 0;
    for str_load in str_loads.iter() {
        let loads: Vec<&str> = str_load.split("\n").collect();
        let mut elf_load = 0;
        for load in loads.iter() {
            elf_load += load.parse::<u32>().unwrap();
        }
        if elf_load > most {
            most = elf_load;
        }
    }
    println!("{:?}", most)
}
